parsesftpdump.py

```diff
@@ -7,30 +7,65 @@
 import re
 from airflow.models import Variable
 
-def parse_sftpdump(ds, **kwargs):
-    UTC_DATESTAMP_FSTR = '%Y-%m-%dT%H:%M:%SZ'
-    marcfile_prefix = 'alma_bibs__'
-    marcfile_extension = 'xml'
-    ingest_command = Variable.get("AIRFLOW_HOME") + "/dags/cob_datapipeline/scripts/ingest_marc_multi.sh"
-    marcfilepath = Variable.get("AIRFLOW_DATA_DIR") + "/sftpdump/"
+UTC_DATESTAMP_FSTR = '%Y-%m-%dT%H:%M:%SZ'
+MARCFILE_PREFIX = 'alma_bibs__'
+MARCFILE_EXTENSION = 'xml'
+ingest_command = Variable.get("AIRFLOW_HOME") + "/dags/cob_datapipeline/scripts/ingest_marc_multi.sh"
+marcfilepath = Variable.get("AIRFLOW_DATA_DIR") + "/sftpdump/"
+
+# parse filename for datestamp
+# e.g. alma_bibs__2018041306_6863237930003811_new_61.xml
+def get_dump_date(marcdircontents):
+    datestamp = None
+    datestampregex = re.compile(r"alma_bibs__(.*)_(.*).xml")
+    marcfileregex = re.compile(MARCFILE_PREFIX + r".*\." + MARCFILE_EXTENSION + "$")
+    for infilename in marcdircontents:
+        if marcfileregex.search(infilename):
+            rxresults = datestampregex.search(infilename)
+            if rxresults is not None:
+                datestamp = rxresults.group(1)
+                break
+    return datestamp
```

Comment:
Reading through: checks for & returns any datestamp found in SFTP files grabbed from the SFTP server

Reply:
what if there are multiple dates present?

Reply:
Then something went wrong and a human has to intervene.